sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -769,6 +770,28 @@ object CombineFilters extends Rule[LogicalPlan] {
 }
 
 /**
+ * Remove all the deterministic conditions in a [[Filter]] that are guaranteed to be true
+ * given the constraints on the child's output.
+ */
+object PruneFilters extends Rule[LogicalPlan] with PredicateHelper {
```

Comment:
Looks like `SimplifyFilters` is similar in purpose with this rule. Can we merge them?